src/octree/batch_iterator.rs

```diff
@@ -91,36 +111,58 @@ where
 
 /// Iterator on point batches
 pub struct BatchIterator<'a> {
-    iterator: Box<Iterator<Item = Point> + 'a>,
+    octree: &'a Octree,
+    culling: PointCulling,
+    local_from_global: Option<Isometry3<f64>>,
     batch_size: usize,
 }
 
 impl<'a> BatchIterator<'a> {
-    pub fn new(
-        octree: &'a octree::Octree,
-        location: &'a octree::batch_iterator::PointLocation,
-        batch_size: usize,
-    ) -> Self {
-        let iterator: Box<Iterator<Item = Point>> = match location {
-            PointLocation::Any() => Box::new(octree.all_points()),
-            PointLocation::Aabb(aabb) => Box::new(octree.points_in_box(aabb)),
-            PointLocation::Frustum(frustum) => Box::new(octree.points_in_frustum(frustum)),
-            PointLocation::OrientedBeam(beam) => Box::new(octree.points_in_oriented_beam(beam)),
+    pub fn new(octree: &'a octree::Octree, location: &'a PointLocation, batch_size: usize) -> Self {
+        let culling = match &location.global_from_local {
+            Some(global_from_local) => match &location.culling {
+                PointCulling::Any() => PointCulling::Any(),
+                PointCulling::Aabb(aabb) => {
+                    PointCulling::Obb(Obb::from(*aabb).transform(global_from_local))
+                }
+                PointCulling::Obb(obb) => PointCulling::Obb(obb.transform(global_from_local)),
+                PointCulling::Frustum(frustum) => PointCulling::Frustum(
+                    Matrix4::from(Decomposed {
+                        scale: 1.0,
+                        rot: global_from_local.rotation,
+                        disp: global_from_local.translation,
+                    }) * frustum,
+                ),
+                PointCulling::OrientedBeam(beam) => {
+                    PointCulling::OrientedBeam(beam.transform(global_from_local))
+                }
+            },
+            None => location.culling.clone(),
         };
+        let local_from_global = location.global_from_local.clone().map(|t| t.inverse());
```

Comment:
what does it happen if location.global_from_local is None?

Reply:
sorry i found it, it maps Option<T> to Option<U

Reply:
`map` leaves the `None` untouched, see https://doc.rust-lang.org/std/option/enum.Option.html#method.map
The point stream internally handles it accordingly.